### ISSUE TITLE
Error building win32a as DLL

### ISSUE THREAD

**Issue by Chiel92** [Bill-Gray/PDCurses#7]
When trying to build win32a as a dll, I get the following error.
Building without the DLL=Y argument works just fine.

```
C:\Users\Chiel.tenBrinke\Projects\PDCurses\win32a>nmake vcwin32.mak WIDE=Y DLL=Y

Microsoft (R) Program Maintenance Utility Version 14.00.23918.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        link  /NOLOGO /DLL /OUT:pdcurses.dll /DEF:pdcursew.def addch.obj addchst
r.obj addstr.obj attr.obj beep.obj  bkgd.obj border.obj clear.obj color.obj delc
h.obj deleteln.obj  deprec.obj getch.obj getstr.obj getyx.obj inch.obj inchstr.o
bj  initscr.obj inopts.obj insch.obj insstr.obj instr.obj kernel.obj  keyname.ob
j mouse.obj move.obj outopts.obj overlay.obj pad.obj  panel.obj printw.obj refre
sh.obj scanw.obj scr_dump.obj scroll.obj  slk.obj termattr.obj terminfo.obj touc
h.obj util.obj window.obj  debug.obj pdcclip.obj pdcdisp.obj pdcgetsc.obj pdckbd
.obj pdcscrn.obj  pdcsetsc.obj pdcutil.obj pdcurses.obj user32.lib gdi32.lib adv
api32.lib shell32.lib comdlg32.lib
pdcurses.obj : fatal error LNK1112: module machine type 'x64' conflicts with tar
get machine type 'X86'
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0
\VC\BIN\link.EXE"' : return code '0x458'
Stop.
```

**Comment by Bill-Gray** [comment]
This one crops up a fair bit... take a look at line 124 of 'vcwin32.mak'.

cvtres /MACHINE:X64 /NOLOGO /OUT:pdcurses.obj pdcurses.res

    Change X64 to IX86,  and you should be in business.  I wish there were
a more straightforward way of dealing with this,  in which the correct
flag were automagically detected and used,  but I don't see how.

-- Bill

**Comment by Chiel92** [comment]
The line now reads `vtres /MACHINE:IX86 /NOLOGO /OUT:pdcurses.obj pdcurses.res`, but I'm getting the exact same error. Yes, I saved the file, and yes I edited vcwin32.mak in the win32a folder, but I guess I still must be forgetting something really stupid.

**Comment by Bill-Gray** [comment]
Um.  That _is_ a bit odd...

    Did you revise that line,  then 'nmake clean',  and then do the build make?
The changed line controls the building of the resource file;  if you changed
vcwin32.mak,  then just ran nmake vcwin32.mak WIDE=Y DLL=Y again,  it wouldn't
cause cvtres to re-run.  Going from a clean make would be needed.

-- Bill

**Comment by Chiel92** [comment]
Ah.

```
C:\Users\Chiel.tenBrinke\Projects\PDCurses\win32a>nmake clean

Microsoft (R) Program Maintenance Utility Version 14.00.23918.0
Copyright (C) Microsoft Corporation.  All rights reserved.

NMAKE : fatal error U1073: don't know how to make 'clean'
Stop.
```

xD

But `nmake /f vcwin32.mak clean` worked like you said!

**Comment by Nadav-Eyesight** [comment]
Brilliant! It works!
I came here after having trouble with the mysterious
`#   NOTE: to make a 32-bit version,  change /MACHINE:X64 to /MACHINE:IX86`

I think it would be helpful if it was changed to 

`#   NOTE: to make a 32-bit version,  change X64 to IX86  at line 124 of 'vcwin32.mak'.`

**Comment by Bill-Gray** [comment]
Thanks for mentioning this.  You'll see that I've modified vcwin32.mak
so one can run 'nmake -f vcwin32.mak (other flags) IX86=Y' to get a 32-bit
build... which is what I should have done in the beginning,  rather than
requiring one to edit vcwin32.mak.

    This caused me to realize that a separate makefile for the Intel(R)
compiler was not really needed;  it's almost identical to the VC/C++
one.  Also,  all of this applies to the console Win32 flavor of PDCurses
about as much as it does to Win32a,  so I updated that version of
'vcwin32.mak' as well.

-- Bill
